client/src/components/notifs/notification.js

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from "react";
+import toast, { Toaster } from "react-hot-toast";
+import { fetchToken, onMessageListener } from "../../firebaseInit";
+import { useLocation } from "react-router-dom";
+
+const Notif = () => {
+  const [notification, setNotification] = useState({ title: "", body: "" });
+  const notify = () =>
+    toast.custom(
+      (t) => (
+        <DisplayToast
+          t={t}
+          title={notification.title}
+          body={notification.body}
+        />
+      ),
+      {
+        duration: 4000,
+        position: "top-right",
+      }
+    );
+
+  useEffect(() => {
+    if (notification.title) {
+      notify();
+    }
+  }, [notification]);
+
+  onMessageListener()
+    .then((payload) => {
+      setNotification({
+        title: payload?.notification?.title,
+        body: payload?.notification?.body,
+      });
+    })
+    .catch((err) => {
+      console.log("failed: ", err);
+    });
+
+  return <Toaster />;
+};
+
+const DisplayToast = (props) => {
+  // console.log("window:", window.location.pathname.split("/"));
+  // const location = useLocation();
+  const path = window.location.pathname.split("/");
+  const showNotif =
+    path[1] !== "chat" || props.title === "You have an invitation";
+
+  return (
+    <>
+      {showNotif ? (
+        <div
+          className={`${
+            props.t.visible ? "animate-enter" : "animate-leave"
+          } max-w-md w-full bg-white shadow-lg rounded-lg pointer-events-auto flex ring-1 ring-black ring-opacity-5`}
+        >
+          <div className="flex-1 w-0 p-4">
+            <div className="flex items-start">
+              <div className="ml-3 flex-1">
+                <p className="text-sm font-medium text-gray-900">
+                  {props.title}
```

Comment:
As a future enhancement we could also mention the chat group name the chat is from in the notification in case the sender belongs to multiple chat groups of the user.

Reply:
Great suggestion! added it.